xbmc/utils/win32/gpu_memcpy_sse4.h

```diff
@@ -0,0 +1,128 @@
+/*
+ *      Copyright (C) 2011-2015 Hendrik Leppkes
+ *      http://www.1f0.de
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ *  Taken from the QuickSync decoder by Eric Gur
+ */
+
+#include <emmintrin.h>
+
+// gpu_memcpy is a memcpy style function that copied data very fast from a
+// GPU tiled memory (write back)
+// Performance tip: page offset (12 lsb) of both addresses should be different
+//  optimally use a 2K offset between them.
+inline void* gpu_memcpy(void* d, const void* s, size_t size)
```

Comment:
On Linux this would prevent the application from loading. For this reason I put the sse4 code into a dll and only load this when sse4 is detected to be available.
https://github.com/xbmc/xbmc/tree/master/xbmc/linux/sse4
Are you sure Windows behaves differently? Did you test on systems without sse4?